src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1659,51 +1659,85 @@ protected override void InvokeDefaultAction(string path)
                             return;
                         }
 
-                        bool attributeFilter = true;
-                        bool switchAttributeFilter = true;
-                        bool filterHidden = false;           // "Hidden" is specified somewhere in the expression
-                        bool switchFilterHidden = false;     // "Hidden" is specified somewhere in the parameters
-
-                        if (null != evaluator)
-                        {
-                            attributeFilter = evaluator.Evaluate(filesystemInfo.Attributes);  // expressions
-                            filterHidden = evaluator.ExistsInExpression(FileAttributes.Hidden);
-                        }
-                        if (null != switchEvaluator)
+                        // Internal test code, run only if the
+                        // 'GciEnumerationActionFilename' test hook is set
+                        var testActionFilename = InternalTestHooks.GciEnumerationActionFilename;
+                        if (filesystemInfo.Name == testActionFilename)
                         {
-                            switchAttributeFilter = switchEvaluator.Evaluate(filesystemInfo.Attributes);  // switch parameters
-                            switchFilterHidden = switchEvaluator.ExistsInExpression(FileAttributes.Hidden);
+                            var fullName = Path.Combine(directory.FullName, filesystemInfo.Name);
+                            var newFilename = InternalTestHooks.GciEnumerationActionRename;
+                            if (String.IsNullOrEmpty(newFilename))
+                            {
+                                File.Delete(fullName);
```

Comment:
This test hook frightens me - please think through the security implications of this.